File: C++/no_of_teams_with_same_color.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int  n;
    cin >> n;
    vector <int> home_codes(101,0);
    vector <int> guest_codes(101, 0);

    int home, guest;
    for(int i=0 ; i<n ; ++i){
        scanf("%d %d", &guest, &home);
        home_codes[home]++;
        guest_codes[guest]++;
    }
    
    int count = 0;
    
    for(int i=1 ; i<=100 ; ++i){
        count += home_codes[i] * guest_codes[i];
    }
    cout << count;
    
    return 0;
}

File: Python/no_of_teams_with_same_color.py
n = int(input(""))
home_codes = [0] * 101
guest_codes = [0] * 101

for i in range(n):
    home, guest = input("").split(" ")
    home = int(home)
    guest = int(guest)
    home_codes[home] += 1
    guest_codes[guest] += 1

count = 0

for i in range(1, 101):
    count += home_codes[i] * guest_codes[i]

print(count)
